src/compute/take/list.rs

```diff
@@ -0,0 +1,214 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::{
+    array::{
+        growable::{Growable, GrowableList},
+        Array, ListArray, Offset, PrimitiveArray,
+    },
+    error::Result,
+};
+
+use super::maybe_usize;
+
+/// `take` implementation for ListArrays
+pub fn take<I: Offset, O: Offset>(
+    values: &ListArray<I>,
+    indices: &PrimitiveArray<O>,
+) -> Result<ListArray<I>> {
+    let usize_indices: Result<Vec<usize>> = indices
+        .values()
+        .iter()
+        .map(|i| maybe_usize::<O>(*i))
+        .collect();
```

Comment:
I _think_ that we can remove this `.collect` and merge this iteration with the iteration below, thus avoiding an extra allocation?

Reply:
I tried that, but I couldn't unwrap the result from the `maybe_usize`. That's why I had to put it in a different allocation.

Reply:
```
    let usize_indices = indices.values().iter().map(|i| maybe_usize::<O>(*i));

    let arrays = usize_indices
        .map(|index| index.map(|x| &(values.slice(x, 1)) as &dyn Array))
        .collect::<Result<Vec<_>>>()?;
```